ui/src/router/components/form/components/router_config/GeneralSettingsPanel.js

```diff
@@ -60,6 +63,31 @@ export const GeneralSettingsPanel = ({
 
         <EuiSpacer size="m" />
 
+        <EuiFormRow
+          fullWidth
+          label={
+            <FormLabelWithToolTip
+              label="Protocol *"
+              content="Specify the type of Turing router Protocol"
+            />
+          }
+          display="row">
+          <EuiSuperSelect
+            fullWidth
+            options={protocolTypeOptions}
+            valueOfSelected={protocol}
+            onChange={(e)=>{
+              onChange("config.protocol")(e)
+              // reset routes and rules when protocol changes
+              onChange("config.routes")([newRoute()])
+              onChange("config.rules")([])
```

Comment:
This seems a bit harsh and the users will lose all their settings (even if only using headers). Should we try to swap "prediction_context" for "payload" and vice versa when the protocol changes but keep everything else in the traffic rules and experiment variables?

Reply:
I thought of it, with some relevance to the earlier convo on the url validation

For the routes, the component values is empty, if it is set it must had passed the url validation regex. 
Just changing the protocol value would had left invalid http endpoint for the UPI router, which is a new state of invalid value set (although it can be caught by yup validation), which lead me to reset it completely.

> Should we try to swap "prediction_context" for "payload" and vice versa when the protocol changes but keep everything else in the traffic rules and experiment variables
Good suggestion, will update accordingly

Reply:
updated

Reply:
updated the videos, around 40s the protocol is changed